### PR TITLE
(RE-14305) Add 'vanagon dependencies' command

### PR DESCRIPTION
**Pull request by isaac-hammes**
Command generates gem dependencies in json files for given projects and platforms in the tmp folder.

### CODE REVIEW

File: lib/vanagon/cli/dependencies.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        VanagonLogger.error e.message
+        exit 1
+      end
+
+      def run(options)
+        platforms_directory = File.join(options[:configdir], 'platforms')
+        projects_directory = File.join(options[:configdir], 'projects')
+
+        unless Dir.exist?(projects_directory) && Dir.exist?(platforms_directory)
+          VanagonLogger.error "Path to #{File.join(options[:configdir], 'platforms')} or #{File.join(options[:configdir], 'projects')} not found."
```
Comment (e-gris) [inline lib/vanagon/cli/dependencies.rb:34]: `#{File.join(options[:configdir], 'platforms')}` => `#{platforms_directory}`

File: lib/vanagon/cli/dependencies.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          exit 1
+        end
+
+        projects = [options[:project_name]]
+        if projects.include?('all')
+          projects = Dir.children(File.join(options[:configdir], 'projects')).map do |project|
```
Comment (e-gris) [inline lib/vanagon/cli/dependencies.rb:40]: `File.join(options[:configdir], 'projects')` => `projects_directory`

There's a few more, I'm sure it makes sense from here

**Review by e-gris — COMMENTED** [review]
There's a couple of rubocop failures to repair then some remaining minor cleanups, below.

File: lib/vanagon/cli/dependencies.rb (hunk 2 above)
Comment (isaac-hammes) [reply lib/vanagon/cli/dependencies.rb:40]: Thanks for these comments and the ones from earlier today. I'll get these fixed tomorrow morning.

**Pull request merged**